### PR TITLE
add T2I-Adapters support

### PR DESCRIPTION
**Pull request by hadipash**

### CODE REVIEW

File: examples/stable_diffusion_v2/adapter_image2image.py [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+    os.environ["SD_VERSION"] = args.version
+    if args.version == "1.5":
+        args.config = "configs/v1-inference.yaml"
+        args.ckpt_path = "models/sd_v1.5-d0ab7146.ckpt"
+    elif args.version == "2.0-v768":
+        args.config = "models/v2-vpred-inference.yaml"
+        args.ckpt_path = "models/sd_v2_768_v-e12e3a9b.ckpt"
+    else:
+        args.config = "configs/v2-inference.yaml"
+        args.ckpt_path = "models/sd_v2_base-57526ee4.ckpt"
```
[the remark this answers is not in the document]
Comment (hadipash) [reply examples/stable_diffusion_v2/adapter_image2image.py:332]: I think the version control should be extracted in a function, so we can just call it from any script and thus reduce code repetitions and errors.

File: examples/stable_diffusion_v2/ldm/models/diffusion/ddim.py [hunk 2]
```diff
@@ -88,6 +88,10 @@ def sample(
         log_every_t=100,
         unconditional_guidance_scale=1.0,
         unconditional_conditioning=None,
+        features_adapter=None,
+        append_to_context=None,
+        cond_tau=0.4,
```
[the remark this answers is not in the document]
Comment (hadipash) [reply examples/stable_diffusion_v2/ldm/models/diffusion/ddim.py:93]: it's configurable through CLI (e.g., `--cond_tau`) but more flexibility with config files is coming soon together with training scripts.

File: examples/stable_diffusion_v2/adapter_image2image.py [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+    logger.info(key_info)
+
+    # infer
+    start_code = None
+    if args.fixed_code:
+        stdnormal = ms.ops.StandardNormal()
+        start_code = stdnormal((args.n_samples, 4, args.H // 8, args.W // 8))
+
+    # TODO: extract into a method
+    cond = cv2.imread(cond, cv2.IMREAD_UNCHANGED)
```
[the remark this answers is not in the document]
Comment (hadipash) [reply examples/stable_diffusion_v2/adapter_image2image.py:119]: Good suggestion. I will research adapters applications in video generation and introduce a better inference strategy in a separate PR then.

File: examples/stable_diffusion_v2/T2I-Adapter.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+| Canny        | 1.5        | [Download](https://download.mindspore.cn/toolkits/mindone/t2i-adapters/t2iadapter_canny_sd15v2-c484cd69.ckpt)    |
+| Color        | 1.4        | [Download](https://download.mindspore.cn/toolkits/mindone/t2i-adapters/t2iadapter_color_sd14v1-7cb31ebd.ckpt)    |
+| Depth        | 1.5        | [Download](https://download.mindspore.cn/toolkits/mindone/t2i-adapters/t2iadapter_depth_sd15v2-dc86209b.ckpt)    |
+| KeyPose      | 1.4        | [Download](https://download.mindspore.cn/toolkits/mindone/t2i-adapters/t2iadapter_keypose_sd14v1-ee27ccf0.ckpt)  |
+| OpenPose     | 1.4        | [Download](https://download.mindspore.cn/toolkits/mindone/t2i-adapters/t2iadapter_openpose_sd14v1-ebcdb5cb.ckpt) |
+| Segmentation | 1.4        | [Download](https://download.mindspore.cn/toolkits/mindone/t2i-adapters/t2iadapter_seg_sd14v1-1d2e8478.ckpt)      |
+| Sketch       | 1.5        | [Download](https://download.mindspore.cn/toolkits/mindone/t2i-adapters/t2iadapter_sketch_sd15v2-6c537e26.ckpt)   |
+| Style        | 1.4        | [Download](https://download.mindspore.cn/toolkits/mindone/t2i-adapters/t2iadapter_style_sd14v1-a620ae97.ckpt)    |
```
[the remark this answers is not in the document]
Comment (hadipash) [reply examples/stable_diffusion_v2/T2I-Adapter.md:34]: Clarified.

**Review by DongyaoZhu — APPROVED** [review]
do we still need controlnet

**Review by SamitHuang — APPROVED** [review]
lgtm

**Pull request merged**